flaskr/logic/adminlgc.py

```diff
@@ -110,10 +111,30 @@ def general_settings(self):
                     else:
                         flash("Interest successfully created")
                         self._interest_handler.insert_interest(current_user.scheme_id, new_interest_form.interest_name.data)
+            
+
+            delete_hobby_form = DeleteHobbyForm(request.form)
+            if delete_hobby_form.hobby.data:
+                delete_id = delete_hobby_form.hobby.data
+                if self._hobby_handler.delete_hobby(current_user.scheme_id, delete_id):
+                    flash("Hobby Removed")
+                else:
+                    flash("Unable To Remove Hobby")
+                
+            delete_interest_form = DeleteInterestForm(request.form)
+            if delete_interest_form.interest.data:
+                delete_id = delete_interest_form.interest.data
+                if self._interest_handler.delete_interest(current_user.scheme_id, delete_id):
+                    flash("Interest Removed")
+                else:
+                    flash("Unable To Remove Interest")
 
-            currentInterests = self._interest_handler.get_interest_list(current_user.scheme_id)
             currentHobbies = self._hobby_handler.get_hobby_list(current_user.scheme_id)
-            return render_template("admin/general_settings.html", hobby_form=new_hobby_form, interest_form=new_interest_form, currentHobbies=currentHobbies, currentInterests=currentInterests)
+            delete_hobby_form.hobby.choices = [(h['id'], h['hobby_name']) for h in currentHobbies]
+            currentInterests = self._interest_handler.get_interest_list(current_user.scheme_id)
+            delete_interest_form.interest.choices = [(i['id'], i['interest_name']) for i in currentInterests]
+            
+            return render_template("admin/general_settings.html", hobby_form=new_hobby_form, delete_hobby_form=delete_hobby_form, delete_interest_form=delete_interest_form, interest_form=new_interest_form, currentHobbies=currentHobbies, currentInterests=currentInterests)
```

Comment:
Should we refactor and make one "General settings" form? 
Instead of 4 smaller ones